torch/distributed/pipelining/schedules.py

```diff
@@ -955,3 +956,151 @@ def backward_stage_index(step):
         for _ in range(self.pp_group_size - rank - 1):
             rank_ops.append(None)
         return rank_ops
+
+
+class ScheduleFlexibleInterleaved1F1B(PipelineScheduleMulti):
+    """
+    The Flexible Interleaved 1F1B schedule.
+    Supports some num_microbatch configurations where num_microbatch % pp_size != 0
```

Comment:
If the only difference is that it supports more num_microbatch configuration sizes, would it make sense to just replace Interleaved1F1B with this implementation? What would be the downside?

Reply:
Synced offline with @H-Huang , we have decided to not merge for now but create shared utility functions for most of the logic.